From 35dd2ed1049a3b954c11b80b023f9679e96b7b1c Mon Sep 17 00:00:00 2001
From: Marc Abramowitz <abramowi@adobe.com>
Date: Thu, 3 Aug 2017 09:11:22 -0700
Subject: [PATCH] Allow --color with no value followed by posarg

Allow commands like `--color features/whizbang.feature` to work
Without this, argparse will treat the positional arg as the value to
--color and we'd get:
  argument --color: invalid choice: 'features/whizbang.feature'
---
 behave/configuration.py | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/behave/configuration.py b/behave/configuration.py
index 1b0bc2b..0fdfd5e 100644
--- a/behave/configuration.py
+++ b/behave/configuration.py
@@ -83,7 +83,7 @@ options = [
 
     (("--color",),
      dict(dest="color", choices=["never", "always", "auto"],
-          default="auto", const="auto", nargs="?",
+          default=None, const="auto", nargs="?",
           help="""Use ANSI color escapes. Defaults to %(const)r.
                   This switch is used to override a
                   configuration file setting.""")),
@@ -562,6 +562,16 @@ class Configuration(object):
             # -- AUTO-DISCOVER: Verbose mode from command-line args.
             verbose = ("-v" in command_args) or ("--verbose" in command_args)
 
+        # Allow commands like `--color features/whizbang.feature` to work
+        # Without this, argparse will treat the positional arg as the value to
+        # --color and we'd get:
+        #   argument --color: invalid choice: 'features/whizbang.feature'
+        #   (choose from 'never', 'always', 'auto')
+        if '--color' in command_args:
+            color_arg_pos = command_args.index('--color')
+            if os.path.exists(command_args[color_arg_pos + 1]):
+                command_args.insert(color_arg_pos + 1, '--')
+
         self.version = None
         self.tags_help = None
         self.lang_list = None
